src/main/scala/codacy/plugins/test/MultipleTests.scala

```diff
@@ -0,0 +1,143 @@
+    val dockerRunner = new BinaryDockerRunner[Result](dockerTool)()
+    val runner = new ToolRunner(dockerTool, toolDocumentation, dockerRunner)
+    val tools = languages.map(new Tool(runner, DockerRunner.defaultRunTimeout)(dockerTool, _))
+    testSources.forall { multipleTestsDir =>
+      val testsDirectories = File(multipleTestsDir).list
+      testsDirectories.forall { testDir =>
+        val srcDir = testDir / "src"
+        val resultFile = testDir / "results.xml"
+        val expectedResults = parseResultsXml(resultFile).toSet
+        Printer.green(s"${testDir.name} should have ${expectedResults.size} results")
+        val configuration = {
+          val patternsPath = testDir / "patterns.xml"
+          if (patternsPath.exists) {
+            val (patterns, extraValues) = parsePatternsXml(patternsPath)
+            if (patterns.isEmpty) FileCfg(Some(srcDir.pathAsString), extraValues)
+            else CodacyCfg(patterns, Some(srcDir.pathAsString), extraValues)
+          } else FileCfg(Some(srcDir.pathAsString), None)
+        }
+        tools.exists { tool =>
+          val res = runTool(tool, srcDir, configuration)
+          res match {
+            case Failure(e) =>
+              Printer.red("Got failure in the analysis:")
+              e.printStackTrace()
+              false
+            case Success(results) =>
+              if (results.sameElements(expectedResults)) {
+                Printer.green(s"Got ${results.size} results.")
+                true
+              } else {
+                Printer.red("Tool results don't match expected results:")
+                Printer.red("Extra: ")
+                pprint.pprintln(results.diff(expectedResults), height = Int.MaxValue)
+                Printer.red("Missing:")
+                pprint.pprintln(expectedResults.diff(results), height = Int.MaxValue)
+                false
+              }
+          }
+        }
+      }
+    }
+  }
+
+  private def parseResultsXml(file: File) = {
+    for {
+      fileTag <- XML.loadFile(file.toJava) \\ "checkstyle" \\ "file"
+      fileName = fileTag \@ "name"
+      errorsTag <- fileTag \\ "error"
+      line = errorsTag \@ "line"
+      patternId = errorsTag \@ "source"
+      message = errorsTag \@ "message"
+      severity = errorsTag \@ "severity"
+      level = severity match {
+        case "info" => Level.Info
+        case "warning" => Level.Warn
+        case "error" => Level.Err
+        case _ => throw new Exception(s"$severity is not a valid level")
+      }
+    } yield PluginResult(patternId, fileName, line.toInt, message, level)
+  }
+
+  private def parsePatternsXml(file: File): (Set[Pattern], Option[Map[String, play.api.libs.json.JsValue]]) = {
+    val rootModule = XML.loadFile(file.toJava)
+    val extraValues = (rootModule \ "property").map { node =>
+      val v = node \@ "value"
+      val value = try {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider using util.Try instead of a try-catch block](https://app.codacy.com/app/Codacy/codacy-plugins-test/pullRequest?prid=4500290)